source/iface/integer-functions.rst

```diff
@@ -7,3 +7,855 @@
+Otherwise, returns ``(x[i] + y[i] + 1) >> 1`` for each element of
+``x`` and ``y``. The intermediate sum does not modulo overflow.
+
+The return type is ``GenInt1`` unless ``GenInt1`` is the
+``__swizzled_vec__`` type, in which case the return type is
+the corresponding ``vec``.
+
+``clamp``
+=========
+
+.. rubric:: Overload 1
+
+::
+
+  template<typename GenInt1, typename GenInt2, typename GenInt3>
+  /*return-type*/ clamp(GenInt1 x, GenInt2 minval, GenInt3 maxval);
+
+Constraints: Available only if all of the following conditions are met:
+
+``GenInt1`` is a generic integer type as defined above;
+
+If ``GenInt1`` is not ``vec`` or the ``__swizzled_vec__`` type, then
+``GenInt2`` and ``GenInt3`` must be the same as ``GenInt1``; and
+
+If ``GenInt1`` is ``vec`` or the ``__swizzled_vec__`` type, then
+``GenInt2`` and ``GenInt3`` must also be ``vec`` or the
+``__swizzled_vec__`` type, and all three must have the same
+element type and the same number of elements.
+
+Preconditions: If the inputs are scalars, the value of ``minval``
+must be less than or equal to the value of ``maxval``. If the
+inputs are not scalars, each ``minval`` must be less than or
+equal to the corresponding ``maxval`` value.
+
+Returns: When the inputs are scalars, returns
+``min(max(x, minval), maxval)``. Otherwise, returns
+``min(max(x[i], minval[i]), maxval[i])`` for each element
+of ``x``, ``minval``, and ``maxval``.
+
+The return type is ``GenInt1`` unless ``GenInt1`` is the
+``__swizzled_vec__`` type, in which case the return type is
+the corresponding ``vec``.
+
+.. rubric:: Overload 2
+
+::
+
+  template<typename NonScalar>
+  /*return-type*/ clamp(NonScalar x, NonScalar::value_type minval,
+                        NonScalar::value_type maxval);
+
+Constraints: Available only if ``NonScalar`` is ``marray``, ``vec``,
+or the ``__swizzled_vec__`` type and is a generic integer type as
+defined above.
+
+Preconditions: The value of ``minval`` must be less than or equal to
+the value of ``maxval``.
+
+Returns: ``min(max(x[i], minval), maxval)`` for each element
+of ``x``.
+
+The return type is ``NonScalar`` unless ``NonScalar`` is
+the ``__swizzled_vec__`` type, in which case the return
+type is the corresponding ``vec``.
+
+``ctz``
+=======
+
+.. rubric:: Overload 1
+
+::
+
+  template<typename GenInt>
+  /*return-type*/ clz(GenInt x);
+
+Constraints: Available only if ``GenInt`` is a generic
+integer type as defined above.
+
+Returns: When the input is a scalar, returns the number of leading 0-bits
+in ``x``, starting at the most significant bit position. Otherwise, returns
+the number of leading 0-bits in each element of ``x``. When a value is 0,
+the computed count is the size in bits of that value.
+
+The return type is ``GenInt`` unless ``GenInt`` is the
+``__swizzled_vec__`` type, in which case the return type is the
+corresponding ``vec``.
+
+.. rubric:: Overload 2
+
+::
+
+  template<typename GenInt>
+  /*return-type*/ ctz(GenInt x);
```

Comment:
I don't think this is an overload and should have its own section/heading.  We have `clz` then `ctz`, which are not overloads.